### PR TITLE
[CL]: Fix oneForZero tick iterator bug

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #XXX

## What is the purpose of the change

This PR fixes the off-by-one in the initial tick iterator state when swapping from asset1 to asset0. It is an amendment to #5469, which was reverted because it mistakenly assumed the error existed for both swap directions.

## Testing and Verifying

The changes are covered in swap-strategy related tests, with additional cases added for both swap directions to ensure different tick spacings are tested.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by p0mvn** [comment]
Could you please breakdown in the PR description more on why this is the correct approach?

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/swapstrategy/one_for_zero.go [hunk 1]
```diff
@@ -191,7 +190,7 @@ func (s oneForZeroStrategy) InitializeNextTickIterator(ctx sdk.Context, poolId u
 // zeroForOneStrategy where we use reverse iterator and have to add one to
```
Comment (mattverse) [inline x/concentrated-liquidity/swapstrategy/one_for_zero.go:190]:
```suggestion
// zeroForOneStrategy where we use reverse iterator.
```

File: x/concentrated-liquidity/swapstrategy/one_for_zero.go [hunk 2]
```diff
@@ -191,7 +190,7 @@ func (s oneForZeroStrategy) InitializeNextTickIterator(ctx sdk.Context, poolId u
...
 // the input. Therefore, we define this method to account for different strategies.
```
Comment (mattverse) [inline x/concentrated-liquidity/swapstrategy/one_for_zero.go:191]:
```suggestion
// Therefore, we define this method to account for different strategies.
```

**Review by mattverse — APPROVED** [review]
LGTM

**Pull request merged**